compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -1081,16 +1085,97 @@ pub fn collect_struct(
 
 #[allow(clippy::too_many_arguments)]
 pub fn collect_enum(
-    _interner: &mut NodeInterner,
-    _def_map: &mut CrateDefMap,
-    _usage_tracker: &mut UsageTracker,
-    _enum_definition: Documented<NoirEnumeration>,
-    _file_id: FileId,
-    _module_id: LocalModuleId,
-    _krate: CrateId,
-    _definition_errors: &mut [(CompilationError, FileId)],
-) -> Option<(StructId, UnresolvedEnum)> {
-    todo!("Implement collect_enum")
+    interner: &mut NodeInterner,
+    def_map: &mut CrateDefMap,
+    usage_tracker: &mut UsageTracker,
+    enum_def: Documented<NoirEnumeration>,
+    file_id: FileId,
+    module_id: LocalModuleId,
+    krate: CrateId,
+    definition_errors: &mut Vec<(CompilationError, FileId)>,
+) -> Option<(TypeId, UnresolvedEnum)> {
+    let doc_comments = enum_def.doc_comments;
+    let enum_def = enum_def.item;
+
+    check_duplicate_variant_names(&enum_def, file_id, definition_errors);
+
+    let name = enum_def.name.clone();
+
+    let unresolved = UnresolvedEnum { file_id, module_id, enum_def };
+
+    let resolved_generics = Context::resolve_generics(
+        interner,
+        &unresolved.enum_def.generics,
+        definition_errors,
+        file_id,
+    );
+
+    // Create the corresponding module for the struct namespace
```

Comment:
Really minor, but...

```suggestion
    // Create the corresponding module for the enum namespace
```

Reply:
Really minor, but objectively correct!